interfaces/builtin/snapd_control.go

```diff
@@ -35,22 +41,42 @@ const snapdControlBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
-// http://bazaar.launchpad.net/~ubuntu-security/ubuntu-core-security/trunk/view/head:/data/apparmor/policygroups/ubuntu-core/16.04/snapd-control
 const snapdControlConnectedPlugAppArmor = `
 # Description: Can manage snaps via snapd.
 
 /run/snapd.socket rw,
 `
 
+type snapControlInterface struct {
+	commonInterface
+}
+
+func (iface *snapControlInterface) SanitizePlug(plug *interfaces.Plug) error {
+	if err := iface.commonInterface.SanitizePlug(plug); err != nil {
+		return err
+	}
+
+	refreshSchedule, ok := plug.Attrs["refresh-schedule"].(string)
+	if ok {
+		if refreshSchedule != "" && refreshSchedule != "managed" {
```

Comment:
I wouldn't  accept refresh-schedule to be set to an empty "" string, it seems strange (easier to leave it just out) and it's not a pattern that our plug/slot rule language supports well